pkg/controller/kubefedcluster/controller.go

```diff
@@ -186,7 +186,6 @@ func (cc *ClusterController) addToClusterSet(obj *fedv1b1.KubeFedCluster) {
 	if err != nil || restClient == nil {
 		cc.RecordError(obj, "MalformedClusterConfig", errors.Wrap(err, "The configuration for this cluster may be malformed"))
 		klog.Errorf("The configuration for cluster %s may be malformed", obj.Name)
-		return
```

Comment:
question: pls, could you explain the reason to remove the return statement here ?

Reply:
The KubeFedCluster controller only does health check (and update status) for those in the `clusterDataMap`. Previously KubeFedClusters with malformed config never have a chance to be added in the `clusterDataMap` just because the `addToClusterSet` function return here, so I remove the return statement in order to let the `updateIndividualClusterStatus` func detect them (by checking whether its `clusterKubeClient` is nil) and update their statuses correctly.